webpack/scenes/ContentViews/Publish/CVPublishFinish.js

```diff
@@ -0,0 +1,147 @@
+import { ExternalLinkAltIcon } from '@patternfly/react-icons';
+import { Redirect } from 'react-router-dom';
+import {
+  selectPublishContentViewsError, selectPublishContentViews,
+  selectPublishContentViewStatus,
+} from './ContentViewPublishSelectors';
+import { selectPublishTaskPoll, selectPublishTaskPollStatus } from '../Details/ContentViewDetailSelectors';
+import getContentViews, { publishContentView } from '../ContentViewsActions';
+import getContentViewDetails, { pollPublishTask, handleFinishedTask } from '../Details/ContentViewDetailActions';
+import Loading from '../../../components/Loading';
+import EmptyStateMessage from '../../../components/Table/EmptyStateMessage';
+import { cvVersionPublishKey } from '../ContentViewsConstants';
+
+
+const CVPublishFinish = ({
+  cvId,
+  checkedItems, setCheckedItems,
+  forcePromote, description, setDescription,
+  setIsOpen, versionCount,
+}) => {
+  const dispatch = useDispatch();
+  const [redirect, setRedirect] = useState(false);
+  const [saving, setSaving] = useState(true);
+  const [polling, setPolling] = useState(false);
+  const [taskErrored, setTaskErrored] = useState(false);
+  const response = useSelector(state => selectPublishContentViews(state, cvId, versionCount));
+  const status = useSelector(state => selectPublishContentViewStatus(state, cvId, versionCount));
+  const error = useSelector(state => selectPublishContentViewsError(state, cvId, versionCount));
+  const pollResponse = useSelector(state =>
+    selectPublishTaskPoll(state, cvVersionPublishKey(cvId, versionCount)), shallowEqual);
+  const pollResponseStatus = useSelector(state =>
+    selectPublishTaskPollStatus(state, cvVersionPublishKey(cvId, versionCount)), shallowEqual);
+
+  useEffect(() => {
+    setSaving(true);
+    dispatch(publishContentView({
+      id: cvId,
+      versionCount,
+      description,
+      environment_ids: checkedItems.map(item => item.id),
+      is_force_promote: (forcePromote.length > 0),
+    }));
+    setDescription('');
+    setCheckedItems([]);
+  }, [dispatch, setSaving, setDescription, setCheckedItems]);
+
+  useEffect(() => {
+    setSaving(true);
+    const { id } = response;
+    if (id && status === STATUS.RESOLVED) {
+      setSaving(false);
+      dispatch(pollPublishTask(cvVersionPublishKey(cvId, versionCount), response));
+      setPolling(true);
+    } else if (status === STATUS.ERROR) {
+      setSaving(false);
+    }
+  }, [JSON.stringify(response), status, error]);
+
+  useEffect(() => {
+    const { state, result } = pollResponse;
+    if (response && (state === 'paused' || result === 'error')) {
+      setTaskErrored(true);
+    }
+    if (response && (state === 'stopped' && result === 'success')) {
+      setPolling(false);
+      dispatch(handleFinishedTask(cvId, cvVersionPublishKey(cvId, versionCount), pollResponse));
+      dispatch(getContentViewDetails(cvId));
+      dispatch(getContentViews);
+      setIsOpen(false);
```

Comment:
When the task finishes, the modal closes completely and we don't get to see the completed donut.  Thoughts on keeping it onscreen for a half second or so?

Reply:
Updated as discussed to stick around for 500 milliseconds..There were extra dispatches in an action I showed you earlier. Removed that and the state update happens without warnings I was seeing earlier with setTimeout.